src/api/app/models/token.rb

```diff
@@ -41,6 +47,30 @@ def package_find_options
   def follow_links?
     package_find_options[:follow_multibuild] || package_find_options[:follow_project_links]
   end
+
+  protected
+
+  def validate_package
```

Comment:
This part is not as simple as that. Each token type has it's own way of finding a package with various options and quirks. Including which names it accepts etc.

Reply:
I moved back the validation of the package parameters (project_name and package_name) and its search to the controller.